optimizely/client/client.go

```diff
@@ -133,6 +133,35 @@ func (o *OptimizelyClient) GetEnabledFeatures(userContext entities.UserContext)
 	return enabledFeatures, nil
 }
 
+func (o *OptimizelyClient) Track(userContext entities.UserContext, eventKey string, eventTags map[string]interface{}) error {
+	if !o.isValid {
+		errorMessage := "Optimizely instance is not valid. Failing GetEnabledFeatures."
+		err := errors.New(errorMessage)
+		logger.Error(errorMessage, nil)
+		return err
+	}
+
+	defer func() {
+		if r := recover(); r != nil {
+			errorMessage := fmt.Sprintf(`Optimizely SDK is panicking with the error "%s"`, string(debug.Stack()))
+			err := errors.New(errorMessage)
```

Comment:
do you want to return that err ? or just created for logger ?

Reply:
it's in a defer

Reply:
it's not going to change the value of the error returned that's why I am asking

Reply:
Ah he's right @thomaszurkan-optimizely, you need to name the variable in the return part of the function signature to `(err error)` so that you can assign `err = errors.New(errorMessage` instead of using `:=`